src/react-native-cool-modals/NativeStackView.js

```diff
@@ -59,6 +51,43 @@ function ScreenView({ colors, descriptors, navigation, route, state }) {
     topOffset,
     transitionDuration,
   } = options;
+  const prevLongFormHeight = usePrevious(longFormHeight);
+  const prevShortFormHeight = usePrevious(shortFormHeight);
+
+  const context = useMemo(
+    () => ({
+      jumpToLong: () => {
+        const screen = findNodeHandle(ref.current);
+        if (screen) {
+          NativeModules.RNCMScreenManager.jumpTo(true, screen);
+        }
+      },
+      jumpToShort: () => {
+        const screen = findNodeHandle(ref.current);
+        if (screen) {
+          NativeModules.RNCMScreenManager.jumpTo(false, screen);
+        }
+      },
+    }),
+    []
+  );
+
+  // When 'autoJumpToNewHeight' option is enabled, automatically "jump" towards either
+  // the new "longFormHeight" or new "shortFormHeight"
+  useEffect(() => {
+    if (autoJumpToNewHeight && longFormHeight !== prevLongFormHeight) {
```

Comment:
hm, I'm not sure actually if it's a good approach. Can we somehow check if it was previously in a longer / shorter state? I think it's okay to automatically adjust the size while it's changing, but I think we should persist if we stay in a longer / shorter position.

I can imagine a case when someone changes the size of the long and the short form at the same time. Hence, it's not clear what's expected behavior. I see it will jump to the long-form regardless of the current state and it's not obvious. I suppose it's not a big deal to make every state transition intentional from the user-side. However, it's still valuable to automatically adjust the size in the same state.